### PR TITLE
Fix overflow of the input field for prompting an LLM

### LINKED ISSUE #313: Unable to send a "Modify with LLM" request with a lengthy prompt due to UI overflow
**Describe the bug**
When the prompt in the input field under "modify with LLM" gets length, it is impossible to send the request because the input field overflows over the send button.

Screenshot 2024-07-29 at 14 19 55

Screenshot 2024-07-29 at 14 25 31

**To Reproduce**
1. Generate tests.
2. Write a lengthy prompt into the input field (e.g., `"use the SupportedLanguage & getPackageFromTestSuiteCode from the codebase"`).
3. Reduce the size of the sidebar with test cases, so that input field overflows on top of the send button (with a long enough prompt you do not even need to reduce the sidebar size; your screen width will be insufficient to render the inut field without it collapsing on the send button).
4. See error: you cannot send a request.

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by arksap2002**
# Description of changes made
When the prompt in the input field under "modify with LLM" gets length, it is impossible to send the request because the input field overflows over the send button.

# Why is merge request needed
*Please write why your merge request is needed*

# Other notes
Closes #313 

- [x] I have checked that I am merging into correct branch

### CODE REVIEW

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt [hunk 1]
```diff
@@ -117,8 +117,7 @@ class TestCasePanelBuilder(
     )
 
     // Create "Remove" button to remove the test from cache
-    private val removeButton =
-        IconButtonCreator.getButton(TestSparkIcons.remove, PluginLabelsBundle.get("removeTip"))
+    private val removeButton = IconButtonCreator.getButton(TestSparkIcons.remove, PluginLabelsBundle.get("removeTip"))
```
Comment (pderakhshanfar) [inline src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt:120]: Why did you change this line?

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt [hunk 2]
```diff
@@ -207,14 +206,11 @@ class TestCasePanelBuilder(
...
             )
-            NotificationGroupManager.getInstance()
-                .getNotificationGroup("Test case copied")
-                .createNotification(
-                    "",
-                    PluginMessagesBundle.get("testCaseCopied"),
-                    NotificationType.INFORMATION,
-                )
-                .notify(project)
+            NotificationGroupManager.getInstance().getNotificationGroup("Test case copied").createNotification(
```
Comment (pderakhshanfar) [inline src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt:209]: Same here

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt [hunk 3]
```diff
@@ -535,18 +535,17 @@ class TestCasePanelBuilder(
...
-                testCase.id,
-                testCase.testName,
-                testCase.testCode,
-                uiContext!!.testGenerationOutput.packageName,
-                uiContext.testGenerationOutput.resultPath,
-                uiContext.projectContext,
-                testCompiler,
-                testsExecutionResultManager,
-            )
+        val newTestCase = TestProcessor(project).processNewTestCase(
```
Comment (pderakhshanfar) [inline src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt:538]: And here

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt [hunk 4]
```diff
@@ -684,7 +683,8 @@ class TestCasePanelBuilder(
      * Updates the current test case with the specified test name and test code.
      */
     private fun updateTestCaseInformation() {
-        testCase.testName = TestAnalyzerFactory.create(language).extractFirstTestMethodName(testCase.testName, languageTextField.document.text)
+        testCase.testName = TestAnalyzerFactory.create(language)
```
Comment (pderakhshanfar) [inline src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt:686]: same comment

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt [hunk 5]
```diff
@@ -304,6 +300,10 @@ class TestCasePanelBuilder(
 
         sendButton.addActionListener { sendRequest() }
 
+        requestComboBox.preferredSize = Dimension(
```
Comment (pderakhshanfar) [inline src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt:303]: Can you add comment about this change?

**Review by pderakhshanfar — CHANGES_REQUESTED** [review]

**Review by Vladislav0Art — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt [hunk 6]
```diff
@@ -304,6 +304,13 @@ class TestCasePanelBuilder(
 
         sendButton.addActionListener { sendRequest() }
 
+        // Set the preferred size of the requestComboBox.
+        // The width is calculated by subtracting the combined heights of languageTextField, requestJLabel,
+        //     sendButton, and a padding of 20 from languageTextField's preferred height.
+        requestComboBox.preferredSize = Dimension(
+            languageTextField.preferredSize.height - requestJLabel.preferredSize.height - sendButton.preferredSize.height - 20,
```
[the remark this answers is not in the document]
Comment (Hello-zoka) [reply src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt:311]: I agree; such implementation looks a lot better. Nitpick: you can even write:
`requestComboBox.preferredSize = Dimension()`

**Review by Vladislav0Art — APPROVED** [review]

**Review by Hello-zoka — APPROVED** [review]

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt (hunk 3 above)
Comment (Vladislav0Art) [reply src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TestCasePanelBuilder.kt]: Outdated. No change here.

**Review by pderakhshanfar — APPROVED** [review]

**Pull request merged**